File: Services/DriverService.cs
using smartPark.Models;
using smartPark.Repository.Interfaces;
using smartPark.Services.Interfaces;



namespace smartPark.Services
{
    public class DriverService : IDriverService
    {
        private readonly IDriverRepository _driverRepository;
        private readonly ILogger<DriverService> _logger;

        public DriverService(IDriverRepository driverRepository, ILogger<DriverService> logger)
        {
            _driverRepository = driverRepository;
            _logger = logger;
        }

        public Driver GetByLicensePlate(string licensePlate)
        {
            return _driverRepository.GetByLicensePlate(licensePlate);
        }

        public Driver GetById(int id)
        {
            return _driverRepository.GetById(id);
        }

        public List<Driver> GetAll()
        {
            return _driverRepository.GetAll();
        }

        public int Add(Driver driver)
        {
            return _driverRepository.Add(driver);
        }

        public void Update(Driver driver)
        {
            _driverRepository.Update(driver);
        }

        public void Delete(int id)
        {
            _driverRepository.Delete(id);
        }
    }
}

File: Client/smartpark-client/src/app/core/services/driver.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Driver, DriverInput } from '../models/driver.model';

@Injectable({ providedIn: 'root' })
export class DriverService {
  private http = inject(HttpClient);
  private baseUrl = `${environment.apiUrl}/Driver`;

  getAll(): Observable<Driver[]> {
    return this.http.get<Driver[]>(this.baseUrl);
  }

  getById(id: number): Observable<Driver> {
    return this.http.get<Driver>(`${this.baseUrl}/${id}`);
  }

  getByLicensePlate(licensePlate: string): Observable<Driver> {
    return this.http.get<Driver>(`${this.baseUrl}/plate/${encodeURIComponent(licensePlate)}`);
  }

  add(driver: DriverInput): Observable<number> {
    return this.http.post<number>(this.baseUrl, driver);
  }

  update(driver: Driver): Observable<void> {
    return this.http.put<void>(this.baseUrl, driver);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
